### PR TITLE
fix: update TypeScript to v5.5

### PR DESCRIPTION
**Pull request by christian-bromann**
## What is the current behavior?
We currently haven't updated to TypeScript 5.5, this patch enables Renovate to trigger the dep update. Since we lowered the update rate, I went ahead and did the update myself and included it in this PR.

STENCIL-1264

## What is the new behavior?
Allow renovate to update to TypeScript 5.5.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

n/a

## Other information

n/a

### CODE REVIEW

File: scripts/esbuild/utils/typescript-source.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ export async function bundleTypeScriptSource(tsPath: string, opts: BuildOptions)
   } catch (e) {}
 
   // get the source typescript.js file to modify
+  console.log('READ', tsPath);
```
Comment (tanner-reits) [inline scripts/esbuild/utils/typescript-source.ts:22]: 🪓

File: scripts/esbuild/utils/typescript-source.ts [hunk 2]
```diff
@@ -74,25 +76,6 @@ export async function bundleTypeScriptSource(tsPath: string, opts: BuildOptions)
...
-  }
-  const lastEnding = code.lastIndexOf(tsEnding);
-  code = code.slice(0, lastEnding);
-
-  const o: string[] = [];
-  o.push(`// TypeScript ${opts.typescriptVersion}`);
-  o.push(code);
-  o.push(`export default ts;`);
-  code = o.join('\n');
-
```
Comment (tanner-reits) [inline scripts/esbuild/utils/typescript-source.ts:95]: Just curious, what changed so we don't need this anymore?

Comment (christian-bromann) [reply scripts/esbuild/utils/typescript-source.ts:95]: The TypeScript team made changes to make it easier consumable for ESM packages (see [announcement](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#easier-api-consumption-from-ecmascript-modules)). I tested it with an `.mjs` file, e.g.
```ts
import * as foo from './node_modules/typescript/lib/typescript.js'

console.log(foo);
```

And it worked great.

Comment (tanner-reits) [reply scripts/esbuild/utils/typescript-source.ts:95]: Nice!

**Review by tanner-reits — APPROVED** [review]

**Pull request merged**